src/components/button/button.js

```diff
@@ -162,20 +162,15 @@ function MdButtonDirective($mdButtonInkRipple, $mdTheming, $mdAria, $timeout) {
     });
 
     if (!element.hasClass('md-no-focus')) {
-      // restrict focus styles to the keyboard
-      scope.mouseActive = false;
-      element.on('mousedown', function() {
-        scope.mouseActive = true;
-        $timeout(function(){
-          scope.mouseActive = false;
-        }, 100);
-      })
-      .on('focus', function() {
-        if (scope.mouseActive === false) {
+
+      element.on('focus', function() {
+        // Only show the focus effect when being focused through keyboard interaction.
+        if ($mdInteraction.getLastInteractionType() === 'keyboard') {
```

Comment:
I just some testing and `mousedown` gets fired after `focus` on Chrome. Instead of tracking mouseactive as a `boolean`, I would suggest a timestamp on last mousedown event. Then attach a timeout to the adding of `md-focused`. If mousedown was triggered within a certain time (say 100ms), don't add the class.